pkg/election/leadership.go

```diff
@@ -152,18 +166,19 @@ func (ls *Leadership) AddCampaignTimes() {
 func (ls *Leadership) Campaign(leaseTimeout int64, leaderData string, cmps ...clientv3.Cmp) error {
 	ls.leaderValue = leaderData
 	// Create a new lease to campaign
-	newLease := &lease{
-		Purpose: ls.purpose,
-		client:  ls.client,
-		lease:   clientv3.NewLease(ls.client),
-	}
-	ls.setLease(newLease)
+	newLease := NewLease(ls.client, ls.purpose)
+	ls.SetLease(newLease)
 
 	failpoint.Inject("skipGrantLeader", func(val failpoint.Value) {
+		name, ok := val.(string)
+		if name == "" {
```

Comment:
Better use `len` which could handle the `nil` case as well.

```suggestion
		if len(name) == 0 {
```